Fix compile error in visual studio 2017 update 5.

RavenX8 opened the pull request:
OS: windows 
Arch: x64
Compiler: VS2017 Community Update 5
Connector: Mysql 5.7 Connector/C
Standard: C++-17
 
Fixes error: 
> sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(90): error C2146: syntax error: missing ')' before identifier 'or'
> sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(92): error C2146: syntax error: missing ')' before identifier 'or'
> sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(97): error C2146: syntax error: missing ')' before identifier 'or'
> sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(103): error C2146: syntax error: missing ')' before identifier 'or'
> sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(109): error C2146: syntax error: missing ')' before identifier 'or'
> sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(115): error C2146: syntax error: missing ')' before identifier 'or'
> sqlpp11-connector-mysql\include\sqlpp11/mysql/connection_config.h(53): error C2146: syntax error: missing ')' before identifier 'and'

This also fixes #37 on all platforms

RavenX8 committed:
Update CMakeLists.txt

Fixed an issue where binaries would get undefined ref errors even though they are linking to the ${MYSQL_LIBRARY}

RavenX8 committed:
add connection_config typedef in connection

RavenX8 committed:
Fixed vs2017 warning

RavenX8 committed:
Merge branch 'master' of https://github.com/RavenX8/sqlpp11-connector-mysql

# Conflicts:
#	include/sqlpp11/mysql/connection.h

Commit:
Updated char_result.h

RavenX8 committed:
Merge pull request #1 from L3nn0x/master

Updated char_result.h

RavenX8 committed:
Fix compile error in visual studio 2017 update 5.

OS: windows 
Arch: x64
Compiler: VS2017 Community Update 5
Connector: Mysql 5.7 Connector/C
Standard: C++-17

Fixes error: ```sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(90): error C2146: syntax error: missing ')' before identifier 'or'
sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(92): error C2146: syntax error: missing ')' before identifier 'or'
sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(97): error C2146: syntax error: missing ')' before identifier 'or'
sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(103): error C2146: syntax error: missing ')' before identifier 'or'
sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(109): error C2146: syntax error: missing ')' before identifier 'or'
sqlpp11-connector-mysql\include\sqlpp11/mysql/char_result.h(115): error C2146: syntax error: missing ')' before identifier 'or'
sqlpp11-connector-mysql\include\sqlpp11/mysql/connection_config.h(53): error C2146: syntax error: missing ')' before identifier 'and'
```

RavenX8 committed:
Merge remote-tracking branch 'upstream/master'

RavenX8 committed:
Added typedef for my_bool if using mysql verison 8

RavenX8 committed:
Fixed using wrong define name for version

RavenX8 committed:
Fix compile error for getting address to temp var

RavenX8 committed:
Merged from master

rbock commented on src/detail/connection_handle.cpp:
```diff
@@ -35,6 +35,10 @@ namespace sqlpp
     namespace detail
     {
+      #if LIBMYSQL_VERSION_ID > 80000
+        typedef bool my_bool;
```
Please use `using my_bool = bool` instead to be in line with the rest of the code.

RavenX8 replied:
Fixed. Also maybe I should move this into it's own header as this block is in 2 files, src/detail/connection_handle.cpp and src/detail/prepared_statement_handle.h.

Let me know what you would prefer.

rbock replied:
Yes, I think it should be in a separate header, like `version_compatibility.h` maybe.

rbock commented on src/prepared_statement.cpp:
```diff
@@ -98,7 +98,7 @@ namespace sqlpp
       param.buffer_length = sizeof(*value);
       param.length = &param.buffer_length;
-      param.is_null = &_handle->stmt_param_is_null[index];
+      param.is_null = (my_bool*)is_null;
```
Why did you change this? c-style casting a bool to a pointer looks rather dangerous to me (even more so, considering that `is_null` is a temporary).

RavenX8 replied:
I had changed this because the compiler was erroring out on that line, saying that 
> &_handle->stmt_param_is_null[index]


was temporary. I don't agree with that statement but this was just a temp change while I tested the rest of the comp process.

RavenX8 replied:
this is the error it throws:

>sqlpp11-connector-mysql/src/prepared_statement.cpp:69:57: error: cannot convert ‘std::vector<bool>::reference* {aka std::_Bit_reference*}’ to ‘bool*’ in assignment
sqlpp11-connector-mysql/src/prepared_statement.cpp: In member function ‘void sqlpp::mysql::prepared_statement_t::_bind_integral_parameter(size_t, const int64_t*, bool)’:
sqlpp11-connector-mysql/src/prepared_statement.cpp:85:57: error: taking address of temporary [-fpermissive]
       param.is_null = &_handle->stmt_param_is_null[index];

RavenX8 replied:
Alright just found out that the reason why is because std::vector can not do what we are doing with booleans. that call instead of returning the address of the bool, it returns the address of a proxy object that IS temporary.

https://en.cppreference.com/w/cpp/container/vector_bool

rbock replied:
Aaargh, yes, it worked before because of `my_bool != bool`.

I am confused by all the `reinterpret_cast`? I think we should just put something different into the vector, like a struct with just a my_bool/bool member.

RavenX8 committed:
Changed typedef to using

reverted temp change of casting is_null

RavenX8 committed:
Changed my_bool from a bool to an int

This was a suggested change by the mysql devs. https://bugs.mysql.com/bug.php?id=85131

RavenX8 committed:
Fix compile error for older libmysql versions

RavenX8 committed:
Merge branch 'master' into patch-1

rbock commented:
I fixed the my_bool part in the meantime.

RavenX8 commented:
Oh so you made the change already? My only question on here was did you want me to move the my_bool def to it's own header file

rbock commented:
Yes, I ran into the problem on travis-ci.

I did not fix the includes for MSVC, though.

RavenX8 commented:
I will be resubmitting this with only the fix for Windows.
